src/lavinmq/queue/queue.cr

```diff
@@ -651,6 +652,7 @@ module LavinMQ
       @queue_expiration_ttl_change.try_send? nil
       @get_count += 1
       get(no_ack) do |env|
+        @basic_get_unacked << env unless no_ack # track unacked messages
```

Comment:
this shoudl proably be done in another place, where we have the channel and deliverytag

Reply:
updated to do this in `channel.basic_get` instead